### ISSUE TITLE
Mypy allows comparing a boolean to a string

### ISSUE THREAD

**Issue by shadycuz** [python/mypy#18047]
**Bug Report**

Mypy does not throw any errors when trying to do something stupid like compare a boolean to string. To be fair this got past flake8 and pylint as well 🙃 

**To Reproduce**

https://mypy-play.net/?mypy=latest&python=3.12&gist=22d4ce8243898e1888b604b9c3171d8d

```python
test = False

if test == "foo":
    print("this aint right")
```

Yes, I sent this into production, but it was a much more complicated example than above. I still wished something would have caught this mistake 😢 

**Expected Behavior**

Some kinda warning that I was about to stay way later at work on a Friday then I had intended

**Actual Behavior**
```console
$ mypy test.py 
Warning: Unpack is already enabled by default
Success: no issues found in 1 source file
```
**Your Environment**

- Mypy version used: 1.10+
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.12

**Comment by brianschubert** [comment]
This is https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-strict-equality

You can use `--strict-equality` (included with [`--strict`](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-strict)) to catch this sort of error.
